doc/prelimsLinux.rst

```diff
@@ -0,0 +1,222 @@
+
+To test that armadillo and its prerequisites have been installed correctly, run the included tester executable with the following commands:
+
+.. code-block:: shell
+
+   cd tests
+   make clean
+   make
+   ./main
+
+c) Install the HDF5 Library
+***************************
+
+The `HDF5 Library <https://support.hdfgroup.org/HDF5/>`__ (compiled with C++ support) is used for storing and managing raw experimental data. It is easiest to install and build the library with CMake, the steps for which can be found `here <https://support.hdfgroup.org/HDF5/release/cmakebuild518.html>`__. To download and uncompress the file, find the link to the most recent release and enter the following into the command line.
+
+.. code-block:: shell
+
+   wget https://support.hdfgroup.org/ftp/HDF5/current18/src/CMake-hdf5-1.8.19.tar.gz
+   tar xzf CMake-hdf5-1.8.19.tar.gz 
+
+Then change to the directory created by the extraction and execute the batch file named *build-unix.sh*.
+
+.. code-block:: shell
+
+   cd CMake-hdf5-1.8.19
+   ./build-unix.sh   # sudo might be required
+
+This will place the built binary in the bin folder and run through a series of tests for correct installation.
+
+d) Install and Compile mcopt Library
+************************************
+
+Finally, install the mcopt library itself; it can be found `here <https://github.com/jbradt/mcopt>`__. Clone the repository locally using the .git link found on GitHub.
+
+.. code-block:: shell
+
+   git clone https://github.com/jbradt/mcopt.git
+   cd mcopt
+
+The compilation and installation instructions can be found in the README.md file in this directory. The necessary commands are as follows.
+
+.. code-block:: shell
+
+   mkdir build && cd build
+   cmake -DCMAKE_BUILD_TYPE=Release ..
+   make
+   make install   # sudo might be required
+
+Test for correct code compilation by running the *test_mcopt* executable:
+
+.. code-block:: shell
+
+   ./test_mcopt
+
+4) Create a new Virtual Env
+---------------------------
+
+Now, create a virtual environment by entering the following into the command line:
+
+.. code-block:: shell
+
+   mkvirtualenv [name]
+
+Refer to the link in step 2 for information on using and managing virtual environments.
+
+
+
+5) Install the pytpc Package
+----------------------------
+
+Now, install the pytpc package and its dependencies; it can be found `here <https://github.com/ATTPC/pytpc.git>`__. Install the repository locally using the .git link found on GitHub.
+
+.. code-block:: shell
+
+   git clone https://github.com/ATTPC/pytpc.git
+   cd pytpc
+
+Installation instructions can be found in the README.md file. However, avoid Anaconda when for pytpc's purposes due to assorted problems with dependency versions and etc. Use pip to manage the required Python software packages.
+
+.. code-block:: shell
+
+   pip3.6 install Cython numpy scipy sklearn scikit-learn matplotlib seaborn jinja2 pandas clint pyYaml sqlalchemy tables h5py sphinx   # sudo might be required
```

Comment:
Note to self: We should really make a `requirements.txt` file. Then you could just say `pip install -r requirements.txt`.